### DISCUSSION TITLE
Close log file

### DISCUSSION THREAD

**Discussion by pagorek18** [gabime/spdlog#2005] [Q&A] [answered]
Hi,
I've created logger like this

```
std::vector<spdlog::sink_ptr> sinks;
sinks.push_back(std::make_shared<spdlog::sinks::stdout_color_sink_st>());
sinks.push_back(std::make_shared<spdlog::sinks::basic_file_sink_st>("filename"));
auto logger = std::make_shared<spdlog::logger>("logger", begin(sinks), end(sinks));
spdlog::register_logger(logger);
```

My build is failing with sonar inspection that opened file is never close. Does anyone can confirm that spdlog will close the file during the `logger` object destruction?

**Comment by tt4g** [comment, marked as answer]
`file_helper` destructor will close the file.

https://github.com/gabime/spdlog/blob/100f30043f33277122e0991c83845a2617172ffd/include/spdlog/details/file_helper-inl.h#L23-L26

Should you call `spdlog::shutdown()` before application exit.
Because this function release all logger from spdlog global registry.
